src/Protocol/Protocol_1_14.cpp

```diff
@@ -104,8 +206,84 @@ void cProtocol_1_14::SendPaintingSpawn(const cPainting & a_Painting)
 
 
 
+void cProtocol_1_14::SendParticleEffect(const AString & a_ParticleName, float a_SrcX, float a_SrcY, float a_SrcZ, float a_OffsetX, float a_OffsetY, float a_OffsetZ, float a_ParticleData, int a_ParticleAmount)
+{
+	ASSERT(m_State == 3);  // In game mode?
+
+	cPacketizer Pkt(*this, pktParticleEffect);
+	Pkt.WriteBEInt32(GetProtocolParticleID(a_ParticleName));
+	Pkt.WriteBool(false);
+	Pkt.WriteBEFloat(a_SrcX);
+	Pkt.WriteBEFloat(a_SrcY);
+	Pkt.WriteBEFloat(a_SrcZ);
+	Pkt.WriteBEFloat(a_OffsetX);
+	Pkt.WriteBEFloat(a_OffsetY);
+	Pkt.WriteBEFloat(a_OffsetZ);
+	Pkt.WriteBEFloat(a_ParticleData);
+	Pkt.WriteBEInt32(a_ParticleAmount);
```

Comment:
This seems to be exactly the same code as the method it overrides

Reply:
It's indeed, the same code but the call to GetProtocolParticleID in the 1.14 protocol returns the particle in 1.14 protocol. While in the 1.13 protocol class, it's calling GetProtocolParticleID of the 1.13 protocol. Maybe I misunderstood that ?

Reply:
Hm. I'm not 100% percent sure about inheritance in this case. Anyway you already have a positive review on your PR so that should be fine

Reply:
Yep, as the GetProtocolParticleID is static rather than virtual the implementation of this function needs to be copied. Ideally the function would become virtual if it needs to be used in this way.